lib/rails_multisite/connection_management.rb

```diff
@@ -121,14 +125,28 @@ def self.host(env)
       end
     end
 
-    attr_reader :config_filename
+    def self.handler_key(spec)
+      if ActiveRecord::Base.respond_to?(:writing_role)
+        :"#{spec.name}_#{ActiveRecord::Base.writing_role}"
```

Comment:
Appending `_writing` postfix here to keep some sort of convention with the default `writing` and `reading` role in Rails. In Rails 6.1, this will not matter because we will do connection pool switching instead of connection handler.